itest/psbt_test.go

```diff
@@ -1608,6 +1619,436 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+	// commit to any outputs.
+	vPkt.Inputs[0].SighashType = txscript.SigHashNone
+
+	// Let's do some sanity checks on the structure of the vPSBT, and prepare
+	// its outputs before signing.
+	require.Equal(t.t, vPkt.Outputs[0].Type, tappsbt.TypeSimple)
+	require.NoError(t.t, tapsend.PrepareOutputAssets(ctxb, vPkt))
+	require.Nil(t.t, vPkt.Outputs[0].Asset.SplitCommitmentRoot)
+	require.Len(t.t, vPkt.Outputs[0].Asset.PrevWitnesses, 1)
+	require.Nil(t.t, vPkt.Outputs[0].Asset.PrevWitnesses[0].SplitCommitment)
+
+	fundedPsbtBytes, err := tappsbt.Encode(vPkt)
+	require.NoError(t.t, err)
+
+	// Alice signs the vPSBT.
+	signedResp, err := alice.SignVirtualPsbt(
+		ctxb, &wrpc.SignVirtualPsbtRequest{
+			FundedPsbt: fundedPsbtBytes,
+		},
+	)
+	require.NoError(t.t, err)
+	require.Contains(t.t, signedResp.SignedInputs, uint32(0))
+
+	// Deserialize the signed vPSBT.
+	vPkt, err = tappsbt.Decode(signedResp.SignedPsbt)
+	require.NoError(t.t, err)
+
+	// Now we need to create the bitcoin PSBT where the previously created
+	// vPSBT will be anchored to.
+	btcpsbt, err := tapsend.PrepareAnchoringTemplate([]*tappsbt.VPacket{
+		vPkt,
+	})
+	require.NoError(t.t, err)
+
+	// This bitcoin PSBT should have 1 input, which is the anchor of Alice's
+	// assets, and 2 outputs that correspond to Alice's anchor that carries the
+	// transfered assets, and one output that carries the bitcoin change.
+	require.Len(t.t, btcpsbt.Inputs, 1)
+	require.Len(t.t, btcpsbt.Outputs, 2)
+
+	// Let's set an actual address for Alice's output.
+	addrResp := t.lndHarness.Alice.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_TAPROOT_PUBKEY,
+	})
+
+	aliceP2TR, err := btcutil.DecodeAddress(
+		addrResp.Address, harnessNetParams,
+	)
+	require.NoError(t.t, err)
+
+	alicePkScript, err := txscript.PayToAddrScript(aliceP2TR)
+	require.NoError(t.t, err)
+
+	// These are basically Alice's terms that she signs the assets over:
+	// Send me 69420 satoshis to this address that belongs to me, and you
+	// will get assets in return.
+	btcpsbt.UnsignedTx.TxOut[0].PkScript = alicePkScript
+	btcpsbt.UnsignedTx.TxOut[0].Value = 69420
+	derivation, trDerivation := getAddressBip32Derivation(
+		t.t, addrResp.Address, t.lndHarness.Alice,
+	)
+	btcpsbt.Outputs[0].Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+	btcpsbt.Outputs[0].TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{
+		trDerivation,
+	}
+	btcpsbt.Outputs[0].TaprootInternalKey = trDerivation.XOnlyPubKey
+
+	var b bytes.Buffer
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// Now we need to commit the vPSBT and PSBT, creating all the related
+	// proofs for this transfer to be valid.
+	resp, err := alice.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{signedResp.SignedPsbt},
+			AnchorPsbt:   b.Bytes(),
+			AnchorChangeOutput: &wrpc.CommitVirtualPsbtsRequest_Add{
+				Add: true,
+			},
```

Comment:
Yes, this is essentially asking for a btc change output to be created.

Previously we checked for 2 outputs because one was Alice's btc "reward" (index `0`) and one was the anchor of the assets

Now that Bob is bringing in his bitcoin, he might want to add an explicit change at index `2`, since after paying alice at index`0` there may be change from his inputs. He could "merge" the btc change with the assets anchor. I believe keeping them separate is more clean and simple

Reply:
Thank you for the explanation.

Given what you said, I think the code comment above is incorrect, it says:
```
	// This bitcoin PSBT should have 1 input, which is the anchor of Alice's
	// assets, and 2 outputs that correspond to Alice's anchor that carries the
	// transfered assets, and one output that carries the bitcoin change.
```
but I think it should say:
```
	// This bitcoin PSBT should have 1 input, which is the anchor of Alice's
	// assets, and 2 outputs that correspond to the BTC payment that Alice
        // will claim (index 0), and Alice's anchor that carries the transferred
        // assets (index 1).
```